integreat_cms/cms/utils/zammad.py

```diff
@@ -0,0 +1,243 @@
+"""
+Zammad API helper functions
+"""
+
+from abc import abstractmethod
+from typing import TYPE_CHECKING
+
+import requests
+from django.conf import settings
+from django.core.cache import cache
+
+if TYPE_CHECKING:
+    from ..models import Region
+
+
+class ZammadAPI:
+    """
+    Zammad API Wrapper. This is intended to be used as a UserChat parent class.
+    """
+
+    @property
+    @abstractmethod
+    def zammad_id(self) -> int:
+        """
+        Property that has to be defined in child classes
+        """
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def region(self) -> "Region":
+        """
+        Property that has to be defined in child classes
+        """
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def device_id(self) -> str:
+        """
+        Property that has to be defined in child classes
+        """
+        raise NotImplementedError
+
+    def zammad_request(
+        self,
+        method: str,
+        path: str,
+        payload: dict | None = None,
+    ) -> requests.Response:
+        """
+        Wrapper for calling the Zammad API. Mostly takes care of auth and timeout.
+
+        :param method: HTTP Method
+        :param path: API path that will be called
+        :param payload: JSON payload as dict
+        :return: Response from Zammad API
+        """
+        response = requests.request(
+            method=method,
+            url=f"{self.region.zammad_url}{path}",
+            timeout=5,
+            headers={"Authorization": f"Token token={self.region.zammad_access_token}"},
+            json=payload,
+        )
+        if response.status_code == 403:
+            raise ValueError("Zammad access token is not valid.")
+        if response.status_code not in [200, 201]:
+            raise ValueError(
+                f"Unexpected response when sending HTTP {method} to "
+                f"{self.region.zammad_url}{path}: {response.status_code}"
+            )
+        return response
+
+    def get_zammad_user_mail(self) -> str:
+        """
+        Get Zammad user e-mail
+
+        :param region: region that is connected to the Zammad server
+        :return: User e-mail address
+        """
+        return self.zammad_request("GET", "/api/v1/users/me").json()["login"]
+
+    def get_zammad_ticket_messages(self) -> list[dict]:
+        """
+        Get Zammad ticket articles
+
+        :return: list of Zammad articles (chat messages)
+        """
+        return self.zammad_request(
+            "GET",
+            f"/api/v1/ticket_articles/by_ticket/{self.zammad_id}",
+        ).json()
+
+    @property
+    def messages(self) -> list[dict]:
+        """
+        Return all messages stored in Zammad for this ticket
+
+        :return: formatted chat messages
+        """
+        if (response := cache.get(f"{self.region.slug}_{self.device_id}")) is None:
+            response = self.get_zammad_ticket_messages()
```

Comment:
I remember from the original implementation that we wantede to not store *any* user messages - has this changed? Do we need user consent for this?

Reply:
Right, maybe I misunderstood from the code - messages are still never cached on our servers?